lxd/device/unix_hotplug.go

```diff
@@ -149,29 +148,36 @@ func (d *unixHotplug) Start() (*deviceConfig.RunConfig, error) {
 	runConf := deviceConfig.RunConfig{}
 	runConf.PostHooks = []func() error{d.Register}
 
-	device := d.loadUnixDevice()
-	if d.isRequired() && device == nil {
+	devices := d.loadUnixDevices()
+	if d.isRequired() && devices == nil {
 		return nil, fmt.Errorf("Required Unix Hotplug device not found")
 	}
 
-	if device == nil {
+	if devices == nil {
 		return &runConf, nil
 	}
 
-	devnum := device.Devnum()
-	major := uint32(devnum.Major())
-	minor := uint32(devnum.Minor())
+	for _, device := range devices {
+		devnum := device.Devnum()
+		major := uint32(devnum.Major())
+		minor := uint32(devnum.Minor())
+
+		// Setup device.
+		var err error
+		if device.Subsystem() == "block" {
+			err = unixDeviceSetupBlockNum(d.state, d.inst.DevicesPath(), "unix", d.name, d.config, major, minor, device.Devnode(), false, &runConf)
+		} else {
+			err = unixDeviceSetupCharNum(d.state, d.inst.DevicesPath(), "unix", d.name, d.config, major, minor, device.Devnode(), false, &runConf)
+		}
 
-	// setup device
-	var err error
-	if device.Subsystem() == "block" {
-		err = unixDeviceSetupBlockNum(d.state, d.inst.DevicesPath(), "unix", d.name, d.config, major, minor, device.Devnode(), false, &runConf)
-	} else {
-		err = unixDeviceSetupCharNum(d.state, d.inst.DevicesPath(), "unix", d.name, d.config, major, minor, device.Devnode(), false, &runConf)
-	}
+		if err != nil {
+			err := unixDeviceRemove(d.inst.DevicesPath(), "unix", d.name, "", &runConf)
```

Comment:
Please can you use the `revert` package here.

This makes it easier to maintain in the future because we wont have to worry about reverting every step manually on any new error cases.

Reply:
Sounds good